starknet_py/net/client_models.py

```diff
@@ -149,24 +149,26 @@ class Transaction(ABC):
     Dataclass representing common attributes of all transactions.
     """
 
-    # Technically, RPC specification moved 'transaction_hash' out of the TXN object, but since it is always returned
-    # together with the rest of the data, it remains here (but is still Optional just in case as spec says)
-    hash: Optional[int]
+    type: str
+    sender_address: int
+    version: str
     signature: List[int]
-    version: int
+    class_hash: int
 
     def __post_init__(self):
         if self.__class__ == Transaction:
             raise TypeError("Cannot instantiate abstract Transaction class.")
 
 
 @dataclass
-class DeprecatedTransaction(Transaction):
+class DeprecatedTransaction(ABC):
     """
     Dataclass representing common attributes of transactions v1 and v2.
     """
 
+    hash: Optional[int]
```

Comment:
I was not sure if I could modify this Deprecated classes without breaking anything... they are being used in other parts of the code... but the required changes in schemas did not include this things like "hash"... so what I did was to remove the usage of this deprecated classes (in my schemas) and create new classes. But as I say above, since I don't know the usage of this, I just moved the old fields in here and make it a base class (not inheriting from the modified transaction class), so in the end, the schema for this will be the same as before. Not sure if that's ok.

Reply:
These changes will cause some breaks for sure, mostly in the tests. I see the following options:
- We can limit the scope of this PR only to changes in the schemas without modifications in the `client_models`
- You can try to take on everything, including edits in the `client_models` that'll cause breaks. If adapting the code to the new changes gets too complex, I can take over from there

Please let me know what you'd prefer.